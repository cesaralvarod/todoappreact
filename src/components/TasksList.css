ul.tasks-list {
  list-style: none;
  margin-top: 2rem;
  padding-top: 1rem;
}
