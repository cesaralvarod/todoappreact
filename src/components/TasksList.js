import React, { useContext } from "react";

// Components

import TaskItem from "./TaskItem";

// Context

import TaskContext from "../helpers/TaskContext";

// Styles

import "./TasksList.css";

const TasksList = ({ tasks }) => {
  const { setArrData } = useContext(TaskContext);

  const handleToggle = (id) => {
    setArrData({ type: "change", id: id });
  };

  const handleDelete = (id) => {
    setArrData({ type: "delete", id: id });
  };

  const tasksRender = tasks.map((task) => {
    return (
      <TaskItem
        key={task.id}
        dataTask={task}
        handleToggle={handleToggle}
        handleDelete={handleDelete}
      />
    );
  });

  return <ul className="tasks-list">{tasksRender}</ul>;
};

export default TasksList;
