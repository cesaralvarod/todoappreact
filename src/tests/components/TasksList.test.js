import React from "react";
import {  mount } from "enzyme";
import TasksList from "../../components/TasksList";
import { demoData } from "../fixtures/demoData";
import TaskContext from "../../helpers/TaskContext";

describe("Testing in component TaskItem", () => {
  const TestComponent = () => (
    <TaskContext.Provider value={{ setArrData() {} }}>
      <TasksList tasks={demoData.data} />
    </TaskContext.Provider>
  );

  const wrapper = mount(<TestComponent />);

  test("Component TaskItem should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Component TaskItem should render 3 TaskItem", () => {
    expect(wrapper.find("TaskItem").length).toBe(demoData.data.length);
  });
});
