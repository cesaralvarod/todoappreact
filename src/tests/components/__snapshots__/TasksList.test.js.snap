// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in component TaskItem Component TaskItem should render correctly 1`] = `
<ul
  className="tasks-list"
>
  <li
    className="task-item"
  >
    <input
      defaultChecked={false}
      id={0}
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="task-content"
      htmlFor={0}
    >
      <span
        className="task-checkbox"
      />
      <p>
        Learn react
      </p>
    </label>
    <button
      className="task-times"
      onClick={[Function]}
    >
      ×
    </button>
  </li>
  <li
    className="task-item"
  >
    <input
      defaultChecked={true}
      id={1}
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="task-content"
      htmlFor={1}
    >
      <span
        className="task-checkbox"
      />
      <p>
        Do homework
      </p>
    </label>
    <button
      className="task-times"
      onClick={[Function]}
    >
      ×
    </button>
  </li>
  <li
    className="task-item"
  >
    <input
      defaultChecked={false}
      id={2}
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="task-content"
      htmlFor={2}
    >
      <span
        className="task-checkbox"
      />
      <p>
        Watch tv
      </p>
    </label>
    <button
      className="task-times"
      onClick={[Function]}
    >
      ×
    </button>
  </li>
</ul>
`;
